### PR TITLE
Refactor linter cli execution and add parsing unittests

### PR DESCRIPTION
**Pull request by awcrosby**
- Refactor parse methods for linters
- Fix error_id parsing for flake8 
- Simplify test calls to ansible-lint .yaml files 
- Add unittests for each linter parse method 

Addresses issues noted in #1689

### CODE REVIEW

File: galaxy/importer/tests/test_linters.py [hunk 1]
```diff
@@ -69,6 +74,42 @@ def test_flake8_fail():
         ]
         expected = [s.format(fp.name) for s in expected]
     assert sorted(result) == sorted(expected)
+    assert 'W504' in error_ids
```
Comment (cutwater) [inline galaxy/importer/tests/test_linters.py]:
```
assert {'W504', 'E117', F821'}.issubset(error_ids)
```

File: galaxy/importer/linters/__init__.py [hunk 2]
```diff
@@ -99,21 +96,17 @@ def _check_files(self, paths):
...
-            msg_parts = message.split(' ')
-            rule_desc = ' '.join(msg_parts[2:])
-
-            error_id = 'YAML_{}'.format(msg_parts[1][1:-1]).upper()
-            if error_id not in ['YAML_ERROR', 'YAML_WARNING']:
-                error_id = None
-
-        except IndexError:
-            error_id = None
+            _, error_level, rule_desc = message.split(' ', 2)
```
Comment (cutwater) [inline galaxy/importer/linters/__init__.py]: I think yamllint errors can be also parsed by single regular expression.

File: galaxy/importer/linters/__init__.py [hunk 3]
```diff
@@ -141,18 +134,13 @@ def _check_files(self, paths):
...
-
-        except IndexError:
-            error_id = None
-
-        if not error_id:
-            logger.error('No error_id found in {} message'.format(self.cmd))
-            return None, None
-
-        return error_id, rule_desc
+            _, error_id, rule_desc = message.split(' ', 2)
```
Comment (cutwater) [inline galaxy/importer/linters/__init__.py]: And ansible-lint errors too.

File: galaxy/importer/tests/test_linters.py [hunk 4]
```diff
@@ -69,6 +74,42 @@ def test_flake8_fail():
...
+
+
+def test_flake8_error_id_match():
+    linter = linters.Flake8Linter()
+    msg_e501 = 'file.py:99:80: E501 line too long (81 > 79 characters)'
+    msg_f401 = 'forms.py:5:1: F401 "pyaml" imported but unused'
+    msg_w191 = 'plugin.py:7:2: W191 indentation contains tabs'
+
+    error_id, _ = linter.parse_id_and_desc(msg_e501)
+    assert error_id == 'E501'
```
Comment (cutwater) [inline galaxy/importer/tests/test_linters.py]: You're testing `parse_id_and_desc` function here. Maybe it makes sense to check if message is parsed correctly as well?

**Comment by chouseknecht** [comment]
Let's hold this for 3.3

**Pull request closed without merging**